scripts/constituents.py

```diff
@@ -100,30 +117,462 @@ def find_variable(self, standard_name=None, source_var=None,
+            cap.write("", 0)
+        # end for
+        cap.write("if (errflg == 0) then", 2)
+        stmt = "call {}%lock_table(ncols, num_layers, num_interfaces, {})"
+        cap.write(stmt.format(const_obj_name, err_callstr), 3)
+        cap.write("end if", 2)
+        cap.write("! Set the index for each active constituent", 2)
+        cap.write("do index = 1, SIZE({})".format(const_indices_name), 2)
+        stmt = "field_ind = {}%field_index({}(index), {})"
+        cap.write(stmt.format(const_obj_name, const_names_name, err_callstr), 3)
+        cap.write("if (field_ind > 0) then", 3)
+        cap.write("{}(index) = field_ind".format(const_indices_name), 4)
+        cap.write("else", 3)
+        cap.write("errflg = 1", 4)
+        stmt = "errmsg = 'No field index for '//trim({}(index))"
+        cap.write(stmt.format(const_names_name), 4)
+        cap.write("end if", 3)
+        cap.write("if (errflg /= 0) then", 3)
+        cap.write("exit", 4)
+        cap.write("end if", 3)
+        cap.write("end do", 2)
+        cap.write("end {}".format(substmt), 1)
+        # Next, write num_consts routine
+        substmt = "function {}".format(num_const_funcname)
+        cap.write("", 0)
+        cap.write("integer {}({})".format(substmt, errvar_str), 1)
+        cap.write("! Return the number of constituent fields for this run", 2)
+        cap.write("", 0)
+        cap.write("! Dummy arguments", 2)
+        for evar in err_vars:
+            evar.write_def(cap, 2, host, dummy=True, add_intent="out")
+        # end for
+        cap.write("", 0)
+        cap.write("{} = {}%num_constituents({})".format(num_const_funcname,
+                                                        const_obj_name,
+                                                        err_callstr), 2)
+        cap.write("end {}".format(substmt), 1)
+        # Next, write copy_in routine
+        substmt = "subroutine {}".format(copy_in_funcname)
+        cap.write("", 0)
+        cap.write("{}(const_array, {})".format(substmt, errvar_str), 1)
+        cap.write("! Copy constituent field info into <const_array>", 2)
+        cap.write("", 0)
+        cap.write("! Dummy arguments", 2)
+        cap.write("real(kind_phys),    intent(out)   :: const_array(:,:,:)", 2)
+        for evar in err_vars:
+            evar.write_def(cap, 2, host, dummy=True, add_intent="out")
+        # end for
+        cap.write("", 0)
+        cap.write("call {}%copy_in(const_array, {})".format(const_obj_name,
+                                                            err_callstr), 2)
+        cap.write("end {}".format(substmt), 1)
+        # Next, write copy_out routine
+        substmt = "subroutine {}".format(copy_out_funcname)
+        cap.write("", 0)
+        cap.write("{}(const_array, {})".format(substmt, errvar_str), 1)
+        cap.write("! Update constituent field info from <const_array>", 2)
+        cap.write("", 0)
+        cap.write("! Dummy arguments", 2)
```

Comment:
I didn't find yet what the second argument to these `write` calls is - number of whitespaces = indent? Ahh I think it is the indent level.

Reply:
Right. More exactly, it is the intent level (indent defaults to 2 spaces).